src/avif.c

```diff
@@ -137,18 +137,12 @@ void avifImageSetDefaults(avifImage * image)
 avifImage * avifImageCreate(uint32_t width, uint32_t height, uint32_t depth, avifPixelFormat yuvFormat)
 {
     // width and height are checked when actually used, for example by avifImageAllocatePlanes().
-    if (depth > 16) {
-        // avifImage only supports up to 16 bits per sample. See avifImageUsesU16().
-        return NULL;
-    }
-    if ((yuvFormat < AVIF_PIXEL_FORMAT_NONE) || (yuvFormat > AVIF_PIXEL_FORMAT_YUV400)) {
-        return NULL;
-    }
+    AVIF_CHECKERR(depth <= 16, NULL); // avifImage only supports up to 16 bits per sample. See avifImageUsesU16().
+    // Cast to silence "comparison of unsigned expression is always true" warning.
+    AVIF_CHECKERR((int)yuvFormat >= AVIF_PIXEL_FORMAT_NONE && yuvFormat < AVIF_PIXEL_FORMAT_COUNT, NULL);
```

Comment:
Since `yuvFormat` is of the `avifPixelFormat` enum type and `AVIF_PIXEL_FORMAT_NONE` is a constant in that enum type, this compiler warning is incorrect.

As reported in https://github.com/AOMediaCodec/libavif/issues/1943, the compiler that warns about this is gcc-4.4.7 on CentOS 6.10. That's a very old version of gcc, and [CentOS 6 reached the end of support in November 2020](https://wiki.centos.org/FAQ(2f)General.html#What_is_the_support_.27.27end_of_life.27.27_for_each_CentOS_release.3F). We don't need to add a workaround for this compiler.

If we want to support this old compiler, please add a comment noting the compiler and OS version, so that we know when it is safe to remove the workaround.